### ISSUE TITLE
Media library problem

### ISSUE THREAD

**Issue by davidnicholls125** [johnvanbreda/naturespot-d8#171]
I've just added an image of a wildlife guide header using the media library and then added a link (using the toolbar as usual) to the guide itself. However the link is just bringing up the header image and not the guide. Can you see what has gone wrong from the html below?

```html
 <a href="https://www.naturespot.org.uk/WildlifeGuide/Lubenham-Main-St-Verge">
 <drupal-media class="img-responsive" data-align="center" data-entity-type="media" data-entity-uuid="a8607591-e239-44f6-a523-f9d8fa0add51"></drupal-media>
 </a>

```

**Comment by johnvanbreda** [comment]
@davidnicholls125 the post didn't show the HTML, but I had rights to edit it - if you edit the post yourself you can see the 3 "apostrophes" (top left key on a Windows keyboard) that can be used to wrap a block of code, and you can then add the language in lowercase to get it to format correctly. 

I found the link you created at the bottom of the ID guides page but it works fine for me and the code looks OK. I tried logged in as admin and also when logged out. Please let me know if you can work out how I can reproduce the issue.

**Comment by davidnicholls125** [comment]
I had deleted the entry after copying the code and did it again with Imce so you must have looked at this working version. I've started again on a test page, re-inserted the image from media library, added the link to the guide and then added the image responsive code. I'm still getting the same problem, when saved the link brings up the inserted image rather than the guide. I've left it this time so you can have a look on the page below.
https://www.naturespot.org.uk/WildPlace/test
(I've tried editing my post as suggested and used the quote button - not sure if this is what you meant?

**Comment by johnvanbreda** [comment]
Ok - I found 2 things going on here. First, I found that the display setting for the full width media view mode was set to "Linked to file" - the setting can be found here: Structure > Media Types > Image > Manage display > Full width (https://www.naturespot.org.uk/admin/structure/media/manage/image/display/full_width). I've removed that setting.

I also found that the text format filter that converts line breaks into HTML paragraphs conflicts with any links placed around media items - described here: https://www.drupal.org/project/drupal/issues/3075527, in particular comment 28. I've turned this filter off for the Full HTML mode which fixes your problem. Actually I suspect this filter is not actually required, as CKEditor will convert your line feeds into paragraphs anyway so it doesn't need to be done again when rendering the content. So, we could remove this filter from the other text formats like Basic HTML, Filtered HTML etc, but if we do there is a chance that some content formatting will be affected.
